api/src/Controller/Stores/BaseProduct.php

```diff
@@ -0,0 +1,58 @@
+<?php declare(strict_types=1);
+
+namespace App\Controller\Stores;
+
+use Atlas\Query\Select;
+
+use Slim\Container;
+use Slim\Http\Request;
+use Slim\Http\Response;
+use App\Controller\BaseController;
+
+abstract class BaseProduct extends BaseController
+{
+    
+    /**
+     * @var int
+     */
+    protected $id = 0;
+
+
+    public function __construct(Container $container)
+    {
+        parent::__construct('products', $container);
+        
+    }
+
+
+    protected function buildProductHateoas(Request $request, Number $store_id)
+    {
+        if ($this->id !== 0) {
+            $path = $request->getUri()->getBaseUrl();
+            $this->addHateoasLink('self', $path.'/stores/'.strval($store_id).'/products/'.strval($this->id), 'GET');
+        }
+
+    }
+
+
+    protected function getProduct(Request $request, Response $response, &$product, $params)
+    {
+        $select = Select::new($this->container->db);
+        $select->columns('ProductID as id', 'StoreID', 'Name', 'ProductSlug', 'Description', 'UnitPriceCents', 'PaymentSystemRef');
+        $select->from('Products')->whereEquals(['ProductID' => $params['id']]);
+        $product = $select->fetchOne();
+
+        if (empty($product)) {
+            return [
+            BaseController::RESULT_TYPE,
+            $this->errorResponse($request, $response, "Could not find Product with ID ${params['id']}", 'Not found', 404)
+            ];
+        }
+
+        return null;
```

Comment:
oh, humm.  I will have to think this design choice.
You are using an argument as a return and returning null.
The rest of the code all returns objects vs arrays and uses is_array and such to determine if it is an error or not.

Reply:
This is actually based direct on `getCycle`, although that's otherwise broken at the moment (it doesn't get request and response passed in, so it will 500 when it reaches the equivalent point.

Reply:
Point being, I assumed there was a Reason we were doing it that way (in `getCycle()`) and followed suit. 

I don't love testing for array vs object as a way of determining error, but I do feel the semantics here should be inverted: return the sought object, or else null, and if null, set `&$error` with the array. Something called `getThing` should return Thing or nothing.